server/build_event_protocol/build_event_handler/build_event_handler.go

```diff
@@ -92,9 +92,10 @@ var (
 )
 
 type BuildEventHandler struct {
-	env           environment.Env
-	statsRecorder *statsRecorder
-	openChannels  *sync.WaitGroup
+	env              environment.Env
+	statsRecorder    *statsRecorder
+	openChannels     *sync.WaitGroup
+	cancelFnsByInvID map[string]func()
```

Comment:
you can run the tests with --@io_bazel_rules_go//go/config:race and it will usually flag stuff like this (if there's a test that exercises it).

In this case I think you will need a mutex because you're writing the map from one goroutine (the RPC request goroutine) and you're reading it from another (the healthchecker shutdown goroutine)

Reply:
Done